components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,528 @@
+        Assert.assertEquals(PRE_ISSUE_ACCESS_TOKEN, action.getType().getActionType());
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 7)
+    public void testUpdateAction() throws ActionMgtException {
+
+        Action updatingAction = buildMockAction(
+                "Pre Issue Access Token",
+                "To configure pre issue access token",
+                "https://sample.com",
+                 AuthType.AuthenticationType.BASIC,
+                 buildMockBasicAuthProperties("updatingadmin", "updatingadmin"));
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+        );
+        action = result;
+    }
+
+    @Test(priority = 8)
+    public void testUpdateActionWithoutNameAndDescription() throws ActionMgtException {
+
+        // TODO: This should be modified once the dao layer is fixed to handle null values.
+        Action updatingAction = buildMockAction(
+                null,
+                null,
+                "https://sample.com",
+                 AuthType.AuthenticationType.BASIC,
+                 buildMockBasicAuthProperties("updatingadmin", "updatingadmin"));
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 9)
+    public void testUpdateActionWithNameAndDescription() throws ActionMgtException {
```

Comment:
Ideally this should fail as well if we are passing null values to the dao for auth type and endpoint as those are mandatory.

Let's add a to do to fix and improve this test case

Reply:
Updated in ecbc2c308facba3438fc42a2ab37e49cb46d1591